### ISSUE TITLE
[WIP] PWM calibration problem

### ISSUE THREAD

**Issue by MaEtUgR** [PX4/PX4-Autopilot#21634]
## Describe the bug
PWM calibration has to work and calibrate between 1000 and 2000us for the 1.14 release. This together with the new actuator configuration, control allocation and https://github.com/PX4/PX4-Autopilot/pull/21513 will give a much better user experience. If it's not all available for the release then people switching to the release and reconfiguring their actuators with the new mixing will have to go through recalibrating and reconfiguring their actuators again.

## Expected behavior
1. Press PWM calibration button or call corresponding command in shell
2. Warning to remove props and instructions to unplug battery (accessible option to abort)
3. After user confirms output exactly 2000us PWM for all motors only
4. Instructions for the user to plug battery/power ESCs (without props!) (accessible option to immediately continue)
5. After user confirms output exactly 1000us PWM for all motors
6. Wait 4 seconds
7. Go back to configured disarmed PWM output
8. Inform the user about the finished process

## To Reproduce
I tested PWM calibration on a pixhwak 4 mini and found multiple issues:
1. The PWM outputs go high immediately after pressing the calibration button even with the battery connected and there is no way to abort again. If I'd have any props on ☠️ 
2. I can't possibly complete the procedure without having an auto-detected battery.
3. Upon ESC calibration timeout the PWM output stays on the high value until I reboot PX4
    - Not debugged yet
4. The highest and lowest value of the PWM calibration corresponds to the configured minimum and maximum PWM value that is used in flight so the idle PWM and the PWM value from which the motor doesn't accelerate anymore instead of 1000 and 2000us.
    - We need to use the consistent, well-defined limits for the calibration case.
5. The battery getting connected is not recognized even though I have a power module that worked in this exact hardware configuration before and its voltage goes above the threshold when I plug the battery but for some reason the board reports zero "valid bricks".
    - Connected stays false because `is_valid()` is false for both ADCs of which the first one would report the correct battery voltage (the second ADC reports 9.5 volts): https://github.com/PX4/PX4-Autopilot/blob/822d7843359bd400494bdf2684df91c75866d712/src/modules/battery_status/analog_battery.cpp#L79-L80
    - "valid bricks" is zero/false for both of the two ADCs because this apparently results in 0 for both lines: https://github.com/PX4/PX4-Autopilot/blob/822d7843359bd400494bdf2684df91c75866d712/boards/px4/fmu-v5/src/board_config.h#L390-L391
    - I quickly checked on a pixracer and on a pixhawk 4 board with main from two weeks ago and they properly detect the battery with an ADC power module while I remember my pixhawk 4 mini had issues with the battery not showing up in the UI already then. So it must be board specific.

**Comment by DronecodeBot** [comment]
This issue has been mentioned on **Discussion Forum for PX4, Pixhawk, QGroundControl, MAVSDK, MAVLink**. There might be relevant details there:

https://discuss.px4.io/t/px4-community-q-a-may-24-2023/32263/1

**Comment by MaEtUgR** [comment]
@dagar I did quite some testing now also with Oneshot and made a fix: https://github.com/PX4/PX4-Autopilot/pull/21711

1. is related to my pixhawk 4 mini setup seeing the battery measurement as present but not detecting battery as `connected` (see 5.) so the calibration is allowed to start which it does immediately after pressing the button in QGC without any additional warning. When battery detection is reliable this problem does not exist. To prevent the worst case I added a safety check in my pr where if theres sudden current consumption measured when pulling the PWM value high it's immediately aborted again.
2. I did not yet get rid of the requirement to have a detectable battery. I'd like to do so and also did successful tests of how to do that but it's a compromise where we start to rely more on timing and the user doing the right thing which would best go with clear instructions in the UI. I could do a parameter where such functionality can be enabled to unblock users that know what they are doing and need to get the ESCs calibrated without battery detection.
3. I found that one. It was just a bug in the procedure. If it failed the output was not released anymore.
4. That's one of the main changes of the PR. The range is now consistent for PWM calibration. Where exactly this is done architecturally should still be changed but this way it works for the 1.14 release.
5. Unresolved. It's for sure something with either just this board or just my setup. The strange thing is that during my testing it very rarely worked as expected 🤔 I think this is not a release blocker.
